### PR TITLE
FEATURE: BB-2569 Sanitize MVF & RF combination

### PR DESCRIPTION
**Pull request by scavnickyj**
When MVF is added to filter bar with existing RF, MVF operator is disabled with warning message.
When RF is added to filter bar with existing MVF, RF attribute dropdown items are disabled.
Separate WarningMessage component with severity prop added.

JIRA: BB-2569

---

Supported PR commands:

| Command         | Description            |
| --------------- | ---------------------- |
| `ok to test`    | Re-run standard checks |
| `extended test` | BackstopJS tests       |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/798/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2386/ : SUCCESS in 12m 43s

File: libs/sdk-ui-filters/src/MeasureValueFilter/Dropdown.tsx [hunk 1]
```diff
@@ -20,12 +21,13 @@ interface IDropdownOwnProps {
...
     value?: IMeasureValueFilterValue;
     usePercentage?: boolean;
-    warningMessage?: string;
+    warningMessage?: WarningMessage;
     locale?: string;
     anchorEl: EventTarget | string;
     separators?: ISeparators;
     displayTreatNullAsZeroOption?: boolean;
     treatNullAsZeroValue?: boolean;
+    isOperatorDisabled?: boolean;
```
Comment (xMort) [inline libs/sdk-ui-filters/src/MeasureValueFilter/Dropdown.tsx:30]: Maybe switch it around? `enableOperatorSelection` with default `true`?
cc  @yanlipnican

File: libs/sdk-ui-filters/src/RankingFilter/AttributeDropdown/DropdownItems/AttributeItem.tsx [hunk 2]
```diff
@@ -46,14 +52,30 @@ export const AttributeItem: React.FC<IAttributeItemProps> = ({
...
-        </button>
+        <BubbleHoverTrigger showDelay={0} hideDelay={0}>
+            <button className={className} disabled={true}>
+                <span>{title}</span>
+            </button>
+            <Bubble
+                className="bubble-primary gd-rf-tooltip-bubble s-rf-attribute-measure-value-filter-used-bubble"
+                alignPoints={[{ align: "cr cl" }, { align: "cl cr" }]}
+            >
+                <FormattedMessage id="rankingFilter.attributeDropdown.measureValueFilterUsed" />
```
Comment (xMort) [inline libs/sdk-ui-filters/src/RankingFilter/AttributeDropdown/DropdownItems/AttributeItem.tsx:77]: This text should not be here IMHO. It should come from AD.

File: libs/sdk-ui-filters/src/RankingFilter/RankingFilterDropdown.tsx [hunk 3]
```diff
@@ -38,6 +38,7 @@ export interface IRankingFilterDropdownProps {
     onDropDownItemMouseOver?: (ref: ObjRefInScope) => void;
     onDropDownItemMouseOut?: () => void;
     anchorEl?: EventTarget | string;
+    isMeasureValueFilterUsed?: boolean;
```
Comment (xMort) [inline libs/sdk-ui-filters/src/RankingFilter/RankingFilterDropdown.tsx:41]: I would change it to `enableCustomGranularitySelection` or something similar.
cc @yanlipnican

Comment (xMort) [reply libs/sdk-ui-filters/src/RankingFilter/RankingFilterDropdown.tsx:41]: Maybe change type to `{ enable: boolean, warningMessage: string }` so you can propagate message from AD and do not presume why it is disabled as there could be other reasons.

Comment (yanlipnican) [reply libs/sdk-ui-filters/src/RankingFilter/RankingFilterDropdown.tsx:41]: customGranularitySelection: { enable: boolean; warningMessage: string; } looks much better

File: libs/sdk-ui-filters/src/MeasureValueFilter/Dropdown.tsx (hunk 1 above)
Comment (yanlipnican) [reply libs/sdk-ui-filters/src/MeasureValueFilter/Dropdown.tsx:30]: I think that operator is usually enabled and people won't disable it very often. So isOperatorDisabled is better.

Comment (yanlipnican) [reply libs/sdk-ui-filters/src/MeasureValueFilter/Dropdown.tsx:30]: but enableOperatorSelection with default true is also great 😄  ... I would say make it consistent and there is already displayTreatNullAsZeroOption and usePercentage  and treatNullAsZeroValue so

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/801/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2389/ : SUCCESS in 26m 00s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/806/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2394/ : SUCCESS in 14m 39s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/812/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2402/ : SUCCESS in 14m 24s

**Comment by scavnickyj** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/813/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/324/ : UNSTABLE in 28m 47s

**Comment by scavnickyj** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/815/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/325/ : UNSTABLE in 37m 10s

**Review by xMort — APPROVED** [review]

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/339/ : SUCCESS in 23s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/595/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2410/ : ABORTED in 32s

**Comment by scavnickyj** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/823/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/328/ : UNSTABLE in 33m 10s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/833/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2420/ : SUCCESS in 17m 21s

**Comment by scavnickyj** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/836/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/333/ : SUCCESS in 18m 26s

**Review by xMort — APPROVED** [review]

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/342/ : SUCCESS in 25s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/598/ : SUCCESS in 15s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2424/ : SUCCESS in 9m 50s

**Pull request merged**